### DISCUSSION TITLE
Performance Issues and Errors with InboxSDK on macOS

### DISCUSSION THREAD

**Discussion by madhavtwig** [InboxSDK/InboxSDK#1200] [Q&A]
Our Chrome extension, which utilizes InboxSDK, is encountering performance issues specifically on macOS. The extension works flawlessly on Windows, but on macOS, it experiences significant slowdowns, Additionally, specific errors related to timeouts are logged.

There is a noticeable slowdown when handling text selection and adding/removing DOM elements.

![Screen Shot 2024-05-27 at 1 45 23 PM](https://github.com/InboxSDK/InboxSDK/assets/152593482/a1b4ec9d-46c5-4ff2-86ef-a26a3713b8e1)

Below is the sample code on how I am using inboxsdk in my content script on gmail:

```

InboxSDK.load(2, "KEY HERE").then(sdk => {
  sdk.Conversations.registerMessageViewHandler(messageView => {

    // Get the subject and body of the email
    subject = messageView.getThreadView().getSubject();
    body = messageView.getBodyElement().innerText;
    
    // Called the API after clicking on a tooltip which appears on text selection
  });
  // the SDK has been loaded, now do something with it!
  sdk.Compose.registerComposeViewHandler(composeView => {
    chrome.storage.local.set({ inboxOpen: true });
    window.composeView = composeView;
    composeView.on("destroy", event => {
      chrome.storage.local.set({ inboxOpen: false });
    });
    composeView.on("restored", event => {
      chrome.storage.local.set({ isReplyOpen: true });
    });
    composeView.addButton({
      title: "Twig AI",
      iconUrl: "https://app.twig.so/twig-ai-icon.png",
      hasDropdown: true,
      onClick(event) {
        event.dropdown.el.innerHTML = `
         2 Buttons here inside UL tag which renders a modal in DOM
        `;
      }
    });
  });
});

```

**Comment by wegry** [comment]
@madhavtwig My initial suspicion here is:

- This issue may be fixed by upgrading to the latest `@inboxsdk/core`
- calling code is the culprit. 

Can you repro with a minimal reproducible example? One way to create an example would be to create a PR where you change one of the existing InboxSDK/InboxSDK example extensions. The example extensions have the added benefit of working source maps.
